### PR TITLE
Praise translators

### LINKED ISSUE #120: Have a section in the site/app to name our awesome translators
It is important that we recognise translation contribution as a key to help us grow internationnaly.

While we have a score board that list all of the contributor to the website, i think we lack a place where people can see their name for the translation work.
Translating OFF is definitly something people should be able to brag about. 

It could well be the only way for people to contribute to a(n open source type of) project at the time.
Translation definitly was my first contribution.

The people did provide me a [place to hang my hat](http://hawthornlandings.org/2015/02/13/a-place-to-hang-your-hat/) in return.
I used it in my CV and I think it we could be the same stepping stone for many other people.

### PR DESCRIPTION
**Pull request by hangy**
Just download the data from the Crowdin API daily with a crontab? We could display the data similarly to https://crowdin.com/project/openfoodfacts/reports#ps-reports-top-members

Would close #120

### CODE REVIEW

**Comment by hangy** [comment]
world productopener localhost_cgi_top_translators pl

**Comment by hangy** [comment]
@stephanegigandet @teolemon Note that this would require some deployment time changes:

- [ ] Update Config2.pm on the servers to contain the correct Crowdin API keys
- [ ] Setup a cronjob that downloads the top translations using `scripts/update_top_translators.pl` once a day

**Comment by teolemon** [comment]
"Vodes made"

File: po/common/common.pot [hunk 1]
```diff
@@ -2680,3 +2688,35 @@ msgstr ""
...
+msgctxt "translators_column_target_words"
+msgid "Target Words"
+msgstr ""
+
+msgctxt "translators_column_approved_words"
+msgid "Approved (Words)"
+msgstr ""
+
+msgctxt "translators_column_votes_made"
+msgid "Vodes Made"
```
Comment (teolemon) [inline po/common/common.pot]: Typo

File: po/common/common.pot [hunk 2]
```diff
@@ -2680,3 +2688,35 @@ msgstr ""
...
+msgctxt "translators_title"
+msgid "Our Translators"
+msgstr ""
+
+msgctxt "translators_lead"
+msgid "We would like to say THANK YOU to the awesome translators that make it possible to present Open Food Facts, Open Beauty Facts, and Open Pet Food Facts to you in all these different languages!"
+msgstr ""
+
+msgctxt "translators_renewal_notice"
+msgid "Please note that this table is renewed nightly and might be out of date."
```
Comment (teolemon) [inline po/common/common.pot]: refreshed nightly

File: po/common/common.pot (hunk 1 above)
Comment (hangy) [reply po/common/common.pot]: Fixed, thanks! :)

File: po/common/common.pot (hunk 2 above)
Comment (hangy) [reply po/common/common.pot]: Changed, thanks! :)

**Comment by stephanegigandet** [comment]
I'm testing it on .net,  the crowdin data is downloaded and the csv is generated, but I'm getting a javascript error:

Failed to load https://static.openfoodfacts.net/data/top_translators.csv: Response to preflight request doesn't pass access control check: The 'Access-Control-Allow-Origin' header contains the invalid value 'static.openfoodfacts.net'. Origin 'https://fr.openfoodfacts.net' is therefore not allowed access.

I need to look into those headers

**Comment by stephanegigandet** [comment]
on .net that's what i have in the apache proxy config (it's different on prod as the proxy is nginx)

SetEnvIf Host "(.+)$" CORS=$0
Header always set Access-Control-Allow-Origin %{CORS}e env=CORS
Header always set Access-Control-Expose-Headers 'Authorization,DNT,User-Agent,Keep-Alive,Content-Type,accept,origin,X-Requested-With,Content-Disposition'
Header always set Access-Control-Allow-Methods "POST, GET, OPTIONS"
Header always set Access-Control-Allow-Credentials "true"

**Comment by stephanegigandet** [comment]
Changing to:

SetEnvIf Host "(.+)$" CORS=$0
SetEnvIf Origin "http(s)?://(.*\.)?(openfoodfacts.org|openfoodfacts.net|openbeautyfacts.org|preprod.openfoodfacts.org|openpetfoodfacts.org)$" CORS=$0
Header always set Access-Control-Allow-Origin %{CORS}e env=CORS


https://stackoverflow.com/questions/1653308/access-control-allow-origin-multiple-origin-domains
https://stackoverflow.com/questions/1653308/access-control-allow-origin-multiple-origin-domains/12414239#12414239

**Comment by hangy** [comment]
Damn CORS 😈 Thanks for testing!

**Comment by stephanegigandet** [comment]
top_translators.pl:1 Failed to load https://static.openfoodfacts.net/data/top_translators.csv: Request header field Range is not allowed by Access-Control-Allow-Headers in preflight response.

even though I did add that header:

Request URL:https://static.openfoodfacts.net/data/top_translators.csv
Request Method:OPTIONS
Status Code:200 OK
Remote Address:87.98.162.131:443
Referrer Policy:no-referrer-when-downgrade

response headers:

Access-Control-Allow-Credentials:true
Access-Control-Allow-Methods:POST, GET, OPTIONS
Access-Control-Allow-Origin:https://fr.openfoodfacts.net
Access-Control-Expose-Headers:Authorization,DNT,User-Agent,Keep-Alive,Content-Type,accept,origin,X-Requested-With,Content-Disposition,range,if-none-match
Allow:OPTIONS,GET,HEAD,POST,TRACE
Connection:Keep-Alive
Content-Length:0
Content-Type:text/plain; charset=ISO-8859-1
Date:Sun, 11 Feb 2018 11:18:17 GMT
Keep-Alive:timeout=2, max=200
Server:Apache

request headers: 

Accept:*/*
Accept-Encoding:gzip, deflate, br
Accept-Language:fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7
Access-Control-Request-Headers:if-none-match,range
Access-Control-Request-Method:GET
Cache-Control:no-cache
Connection:keep-alive
Host:static.openfoodfacts.net
Origin:https://fr.openfoodfacts.net
Pragma:no-cache
User-Agent:Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.119 Mobile Safari/537.36

**Comment by hangy** [comment]
Mabye there's a difference between `Access-Control-Expose-Headers` and `Access-Control-Allow-Headers` https://stackoverflow.com/a/35707821/11963

**Comment by stephanegigandet** [comment]
That was it, thanks @hangy ! https://fr.openfoodfacts.net/cgi/top_translators.pl

**Comment by stephanegigandet** [comment]
Small encoding bug: SimÃ³n RodrÃ­guez

**Comment by hangy** [comment]
Merged after discussion with @stephanegigandet on Slack

**Pull request merged**